### PR TITLE
Add support for channelData field in conversation references

### PR DESCRIPTION
**Pull request by benbrown**
## Description

The conversationReference type is missing fields necessary to use `createConversation` with MS Teams.  This PR adds those fields and amends the functions so that they will work with Teams.

## Specific Changes

Adds the channelData field to the conversation reference.  Information in channelData may be required for operations that use references -- for example, createConversation.

Amends the adapter's createConversation method to pass through the tenant id, which is necessary for using this method with MS Teams.

## Testing

* Connect a bot to Teams.
* Using a conversation reference extracted from an incoming GROUP message, use adapter.createConversation to create a new PRIVATE context.
* Use the new context to send a private message to the user.  Note that it is received as a DM, instead of a reply in the group channel.

### CODE REVIEW

**Comment by cleemullins** [comment]
@benbrown, is there anything we need to pickup on the C# side for this?

File: libraries/botbuilder/src/botFrameworkAdapter.ts [hunk 1]
```diff
@@ -230,6 +230,11 @@ export class BotFrameworkAdapter extends BotAdapter {
         const client: ConnectorClient = this.createConnectorClient(reference.serviceUrl);
         const response = await client.conversations.createConversation(parameters);
 
+        // Mix in the tenant ID if specified. This is required for MS Teams.
+        if (reference.channelData && reference.channelData.tenant) {
```
Comment (vangelov-i) [inline libraries/botbuilder/src/botFrameworkAdapter.ts]: hey, @benbrown , it looks like this snipped of code is supposed to be above the `createConversation` call.
This is how I tweaked it to get it working as intended.
Cheers

File: libraries/botframework-schema/src/index.ts [hunk 2]
```diff
@@ -316,6 +316,12 @@ export interface ConversationReference {
    * the referenced conversation may be performed
    */
   serviceUrl: string;
+
+  /**
+   * @member {any} [channelData] Channel specific payload for re-creating the
+   * conversation
+   */
+  channelData?: any;
```
[the remark this answers is not in the document]
Comment (benbrown) [reply libraries/botframework-schema/src/index.ts]: Shoot! You are right, I added this by hand.

[the remark this answers is not in the document]
Comment (benbrown) [reply libraries/botframework-schema/src/index.ts]: Perhaps this information can go into the `conversationReference.channel` field instead.  However, that would still require a change to the schema. I'll collect some feedback and regroup.

File: libraries/botbuilder/src/botFrameworkAdapter.ts (hunk 1 above)
Comment (benbrown) [reply libraries/botbuilder/src/botFrameworkAdapter.ts]: Thanks for catching this! Fixed!

**Comment by benbrown** [comment]
OK! The service code has changed to include this new field in the schema.

We need to regenerate!

**Comment by benbrown** [comment]
I have manually made the very small schema changes. These should match any future update from swagger.

**Comment by benbrown** [comment]
Just waiting on one final tweak to the dot-net version, then we can merge both!

https://github.com/Microsoft/botbuilder-dotnet/pull/1362#issuecomment-463345903

**Pull request merged**